osu.Game/Screens/Play/Player.cs

```diff
@@ -820,6 +823,40 @@ protected override bool OnScroll(ScrollEvent e)
             return mouseWheelDisabled.Value && !e.AltPressed;
         }
 
+        #region Gameplay leaderboard
+
+        protected readonly Bindable<bool> LeaderboardExpandedState = new BindableBool();
+
+        private void loadLeaderboard()
+        {
+            HUDOverlay.HoldingForHUD.BindValueChanged(_ => updateLeaderboardExpandedState());
+            LocalUserPlaying.BindValueChanged(_ => updateLeaderboardExpandedState(), true);
+
+            LoadComponentAsync(CreateGameplayLeaderboard(), leaderboard =>
+            {
+                if (!LoadedBeatmapSuccessfully)
+                    return;
+
+                leaderboard.Expanded.BindTo(LeaderboardExpandedState);
+
+                AddLeaderboardToHUD(leaderboard);
+            });
+        }
+
+        public readonly BindableList<ScoreInfo> LeaderboardScores = new BindableList<ScoreInfo>();
```

Comment:
I would move this to `SoloPlayer` and `ReplayPlayer` rather than place it here, since not all players are based off a leaderboard with finalised scores.